db/sqlinterfaces.c

```diff
@@ -2944,12 +2944,42 @@ static inline int check_user_password(struct sqlclntstate *clnt)
 /* Return current authenticated user for the session */
 char *get_current_user(struct sqlclntstate *clnt)
 {
-    if (clnt && !clnt->is_x509_user && clnt->have_user) {
-        return clnt->user;
+    if (clnt && !clnt->current_user.is_x509_user &&
+        clnt->current_user.have_name) {
+        return clnt->current_user.name;
     }
     return NULL;
 }
 
+void copy_user(struct user *dst, struct user *src)
+{
+    dst->have_name = src->have_name;
+    dst->have_password = src->have_password;
+    dst->is_x509_user = src->is_x509_user;
+
+    if (dst->have_name) {
+        strcpy(dst->name, src->name);
+    }
+    if (dst->have_password) {
+        strcpy(dst->password, src->password);
+    }
+}
+
+static void reset_user(struct sqlclntstate *clnt)
+{
+    if (!clnt)
+        return;
+
+    /* reset the user */
```

Comment:
how about: 
memset(&clnt->current_user, 0, sizeof(struct user));

Reply:
or maybe :
clnt->current_user.have_name = 0;
    clnt->current_user.is_x509_user = 0;
clnt->current_user.name[0] = '\0';
clnt->current_user.have_password = 0;
clnt->current_user.password[0] = '\0';
...
Not sure which would be faster. Certainly have function be inline for completenes.

Reply:
We could just bzero `current_user` all together.